.github/workflows/validate.yml

```diff
@@ -0,0 +1,61 @@
+name: CI
+on:
+  push:
+    branches: [ main ]
+    tags: ['*']
+
+  pull_request:
+
+permissions:
+  contents: write
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
```

Comment:
So not a huge deal, but you could simplify this a bit by using an os matrix. So instead of having two duplicated jobs apart from the os you could have:

```yml
jobs:
  test:
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ 'ubuntu-latest', 'windows-latest' ]

    steps:
      - name: Checkout Course Management Tools Repo
        uses: actions/checkout@v3
        with:
          fetch-depth: 0

      - name: Set up JVM
        uses: actions/setup-java@v3
        with:
          distribution: 'adopt-hotspot'
          java-version: 11

      - name: Run tests
        run: sbt test
```

And it would create two jobs just like you have, one for each os.

Reply:
Great suggestion. There's one subtle difference between the Windows and the Linux 'version'. In the former, I explicitly specify the shell to be `bash`. IIRC, it's Powershell by default and for some unknown reason, `dev dirs` gets messed up when using Powershell. Will apply your suggestion in a separate step.

Reply:
Just tried that but running into a weird problem; `scalafmt` check fails on Windows.

This suggests that there's a difference of the output generated by `scalafmtAll` between Windows and Linux?

CR/LF perhaps? Maybe there's an option to change this behaviour.